### DISCUSSION TITLE
Unable to find `next-intl` locale because the middleware didn't run on this request.

### DISCUSSION THREAD

**Discussion by arslanovngc** [amannn/next-intl#1263] [Q&A]
### Description

Everything works fine. Here is my middleware 

![image](https://github.com/user-attachments/assets/b55a8647-9ec9-4ed3-887d-ea8363e69eff)

When I run my code and open the page in the browser, the console (inside terminal) is clear. But when I open browser's console and refresh the page "Unable to find `next-intl` locale because the middleware didn't run on this request. See https://next-intl-docs.vercel.app/docs/routing/middleware#unable-to-find-locale. The `notFound()` function will be called as a result." this one error appeared. 

![image](https://github.com/user-attachments/assets/db9bfe67-dc57-4911-8610-68b97bca5e7c)

I tested it in chrome, firefox, microsoft edge and same result. Error is appearing only when browser's console opened

Versions:
"react": "^18.3.1",
"next": "^14.2.4",
"next-intl": "^3.17.2"

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

Sorry, I can not show my code

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Click on …
3. See error: … 

### Expected behaviour

Console should be clear

**Comment by amannn** [comment]
It seems like there's an issue with your matcher regex and sourcemap requests. Review the linked troubleshooting guide in the error message or consider using [the recommended matcher](https://next-intl-docs.vercel.app/docs/routing/middleware#matcher-config) based on locale prefixes.

I'm going to move this to a discussion since this is not a bug report, but a usage question.
